packages/manager/apps/web/client/app/domain/list/list-domain-layout.controller.js

```diff
@@ -94,6 +95,31 @@ export default class ListDomainLayoutCtrl extends ListLayoutHelper.ListLayoutCtr
     };
   }
 
+  getDate(domain, key) {
+    const date = new Date(domain[key]);
+    const locale = this.coreConfig.getUserLocale().replace('_', '-');
+    if (
+      key === 'renewalDate' &&
+      domain.renewalState === this.DOMAIN_RENEWAL_MODE.AUTOMATIC_RENEW
+    ) {
+      const formattedDate = new Intl.DateTimeFormat(locale, {
+        month: '2-digit',
+        year: 'numeric',
+      }).format(date);
```

Comment:
Maybe move the object conf in constant file {
        month: '2-digit',
        year: 'numeric',
      }